### PR TITLE
RPM and Debian packaging fixes

### LINKED ISSUE #530: Request: Ubuntu/Debian Package via Open Build Service
With OpenSuse or CentOS, installation is simple, because they are available in the via the Open Build Service.

```sudo zypper ar http://download.opensuse.org/repositories/devel:/libraries:/ACE:/micro/openSUSE_Tumbleweed/devel:libraries:ACE:micro.repo```

This also, means it's really simple to add in a Dockerfile.

 @jwillemsen is it possible to add Ubuntu?

Sorry if this is the wrong place for a request... But it looks like you're the maintainer of both
https://build.opensuse.org/project/users/devel:libraries:ACE

Thanks!

### PR DESCRIPTION
**Pull request by dpward**
This fixes various issues that appear when building RPM or Debian packages from source distributions of ACE+TAO.

Note that this restores files several files to their standard names — `debian/compat`, `debian/control`, `debian/changelog`, and `debian/rules` — according to the [Debian New Maintainers' Guide](https://www.debian.org/doc/manuals/maint-guide/dreq.en.html).

When releasing the next version of ACE+TAO, please also incorporate a [revision](https://build.opensuse.org/package/revisions/home:davidpward:branches:devel:libraries:ACE:micro/ACE) to this package in the openSUSE Build Service that, combined with the changes in this pull request, allows builds to [succeed](https://build.opensuse.org/project/show/home:davidpward:branches:devel:libraries:ACE:micro) for Debian and Ubuntu. This would fix #530.

### CODE REVIEW

File: ACE/debian/cmp-autoconf-regular.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (jwillemsen) [inline ACE/debian/cmp-autoconf-regular.sh]: This file shouldn't be needed anymore as far as I remember, if so, please remove it

File: ACE/debian/ace.dsc [hunk 2]
```diff
@@ -1,10 +0,0 @@
-Format: 1.0
```
Comment (jwillemsen) [inline ACE/debian/ace.dsc]: Check ACE/bin/make_release.py, I think the script expects this file, also that python script has to be checked for all other file renames, it updates the content from 6.4.7 to 6.4.8 with the next release

File: ACE/debian/control [hunk 3]
```diff
@@ -257,7 +257,7 @@ Description: ACE-GUI reactor integration for Xt
 Package: libace-xtreactor-dev
 Architecture: any
 Section: libdevel
-Depends: libace-xtreactor-6.4.6 (= ${binary:Version}), libace-dev (= ${binary:Version}), libxt-dev (>= 6.4.7), ${misc:Depends}
```
Comment (jwillemsen) [inline ACE/debian/control]: Probably some bugs in bin/make_release.py, that script should update this file

**Comment by jwillemsen** [comment]
I upload each time to OBS after a release, ping me after x.4.8 has been released to integrate your changes there

File: ACE/debian/cmp-autoconf-regular.sh (hunk 1 above)
Comment (dpward) [reply ACE/debian/cmp-autoconf-regular.sh]: I removed this and `ACE/debian/update_file_versions.sh`, which are in the [Debian source repository](https://anonscm.debian.org/cgit/pkg-ace/pkg-ace.git/) but are not needed.

File: ACE/debian/control (hunk 3 above)
Comment (dpward) [reply ACE/debian/control]: There was a regex in `ACE/bin/make_release.py` which was unintentionally matching on the whitespace separating packages on the same line. I fixed this.

**Comment by jwillemsen** [comment]
Thanks for the improvements, when looking at the debian directory it looks the release script isn't updating the content of the lintian-overrides files, could you try to add that?

**Comment by jwillemsen** [comment]
Have you tested the release script, not that we get problems when releasing x.4.8 at some point

**Comment by dpward** [comment]
All of these commands were tested separately:
```
./ACE/bin/make_release.py --micro --update -n
./ACE/bin/make_release.py --micro --update
./ACE/bin/make_release.py --minor --update -n
./ACE/bin/make_release.py --minor --update
./ACE/bin/make_release.py --major --update -n
./ACE/bin/make_release.py --major --update
```

The files in the `ACE/debian` directory with the version number in their name were correctly renamed. A `grep` command did not find the old version number anywhere in this directory (except in the Debian package changelog, which this script was never designed to modify). In `ACE/debian/control`, I saw that the version checks for external dependencies were not inadvertently changed, as they had been during the 6.4.6 release.

My aim here is to fix issues I've identified in the RPM/Debian packaging files and the release script, which are causing those package builds not to work in the way they were originally intended to. There changes to the `ACE/bin/make_release.py` script are constrained to updating the `ACE/debian` directory. The only other considerations with these changes is that they keep things in sync with the Debian source repository, to hopefully avoid duplication of effort; and it will allow the openSUSE Build Service to be configured in a much simpler way that only requires uploading the .tar.gz file in order to build a new release.

**Comment by jwillemsen** [comment]
Thanks for the work done! Could you add instructions what to add/remove/change on OBS with the next x.4.8 release to this issue? Will try to do that directly after the next micro

**Comment by dpward** [comment]
Here are the commands to issue when ACE+TAO x.4.8 is released to update and reconfigure OBS:

```
# Checkout the OBS package without verisoned namespaces

osc checkout devel:libraries:ACE:micro ACE
cd "devel:libraries:ACE:micro/ACE/"

# Replace the ACE+TAO source distribution

osc rm ACE+TAO-src-6.4.7.tar.gz
cp $DOC_ROOT/package-<PID>/ACE+TAO-src-6.4.8.tar.gz .
osc add ACE+TAO-src-6.4.8.tar.gz

# Reconfigure the OBS package (one-time change)

osc rm ace-tao.spec ace-tao-rpmlintrc debian.*

osc mv ace.dsc debian.dsc
cat > debian.dsc << EOF
Format: 3.0 (quilt)
Source: ace
Architecture: any
Version: 0-0
Maintainer: Johnny Willemsen <jwillemsen@remedy.nl>
Standards-Version: 4.1.1
Build-Depends: debhelper, libssl-dev, libxt-dev (>= 4.3.0), libfltk1.1-dev (>= 1.1.4), tk-dev (>= 8.5), libfox-1.6-dev, docbook-to-man, libxerces-c-dev
Build-Depends-Indep: doxygen, graphviz
DEBTRANSFORM-RELEASE: 1
EOF

cat > _service << EOF
<services>
  <service name="extract_file">
    <param name="archive">*.tar.gz</param>
    <param name="files">*/rpmbuild/ace-tao.spec</param>
    <param name="outfilename">ace-tao.spec</param>
  </service>
  <service name="extract_file">
    <param name="archive">*.tar.gz</param>
    <param name="files">*/rpmbuild/ace-tao-rpmlintrc</param>
    <param name="outfilename">ace-tao-rpmlintrc</param>
  </service>
  <service name="extract_file">
    <param name="archive">*.tar.gz</param>
    <param name="files">*/debian/changelog</param>
    <param name="outfilename">debian.changelog</param>
  </service>
  <service name="set_version">
    <param name="basename">ACE+TAO-src</param>
    <param name="file">debian.dsc</param>
  </service>
</services>
EOF
osc add _service

# Commit the changes to the OBS package

osc commit -m "Upgrade to ACE 6.4.8 and TAO 2.4.8; convert package to use OBS services"
cd ../../



# Checkout the OBS package with verisoned namespaces

osc checkout --unexpand-link devel:libraries:ACE:micro:versioned ACE
cd "devel:libraries:ACE:micro:versioned/ACE/"

# Reconfigure the OBS package (one-time change)

osc rm *

cat > _link << EOF
<link project="devel:libraries:ACE:micro" package="ACE">
  <patches>
    <topadd>%define _with_versioned 1</topadd>
  </patches>
</link>
EOF
osc add _link

# Commit the changes to the OBS package

osc commit -m "Convert to linked package"
cd ../../
```

Likewise when ACE+TAO x.5.0 is released, run the same commands above, but substitute _minor_ in place of _micro_; _6.5.0_ in place of _6.4.8_; and _6.4.0_ in place of _6.4.7_.

**Comment by jwillemsen** [comment]
Great! Will do with the next release, thanks for your contributions!

**Comment by jwillemsen** [comment]
New 6.4.8 release has been uploaded to OBS, seems most packages are there, including several debian versions, thanks for your additions!

**Pull request merged**